lib/ansible/plugins/loader.py

```diff
@@ -534,7 +544,13 @@ def get(self, name, *args, **kwargs):
         collection_list = kwargs.pop('collection_list', None)
         if name in self.aliases:
             name = self.aliases[name]
-        path = self.find_plugin(name, collection_list=collection_list)
+        plugin_load_detail = dict()
+        path = self.find_plugin(name, collection_list=collection_list, load_detail_out=plugin_load_detail)
```

Comment:
Oops.... When you were talking about this yesterday, I thought you were going to change the return value to be a dict rather than passing in a dict.  Mea culpa.  I just read what you said you were planning to do wrong.  I don't think operating by side effect, is the best way to write this, especially when we have a side effect and returning a value from the function.

Since we're under time pressure I could see doing it this way for 2.9.x  but I think it should get changed to being the return value in devel.  (Unless it's simple to take what @mattclay did to identify the locations that need to be updated and then use a dist instead of a tuple for all of those).